_includes/sidebar.html

```diff
@@ -2,34 +2,47 @@
   <div class="container sidebar-sticky">
     <div class="sidebar-about">
       <h1>
-        <a href="{{ site.baseurl }}">
-          {{ site.title }}
+        <a href="{{site.baseurl}}">
+          {{site.title}}
         </a>
       </h1>
-      <p class="lead">{{ site.description }}</p>
+      <p class="lead">{{site.description}}</p>
     </div>
 
     <nav class="sidebar-nav">
-      {% for lesson_type in site.sections %}
-      {% assign category = lesson_type | downcase %}
-        <div class="sidebar-nav-group {{ category }}">
+      {% assign lang = page.lang || site.default_lang %}
+      {% for section in site.sections %}
+      {% assign title = section.label | select: lang | first | last %}
+        <div class="sidebar-nav-group {{section.tag}}">
           <div class="sidebar-nav-header">
             <i class="fa fa-chevron-down"></i>
-            <span>{{ lesson_type }}</span>
+            <span>{{title}}</span>
           </div>
           <div class="sidebar-nav-list">
-            {% assign list = site.pages | where: "category", category | sort: "order"  %}
+            
+            {% assign list = site.pages | where: "category", section.tag | where: "lang", lang | sort: "order" %}
             {% for node in list %}
               <div class="sidebar-nav-item">
-                {{ node.order }}. <a class="sidebar-lesson {% if page.url == node.url %} active{% endif %}" href="{{ node.url }}">{{ node.title }}</a>
+                {{node.order}}. <a class="sidebar-lesson {% if page.url == node.url %} active{% endif %}" href="{{node.url}}">{{node.title}}</a>
               </div>
             {% endfor %}
           </div>
         </div>
       {% endfor %}
-
     </nav>
 
+    <p>
+      {% if page.lang == site.default_lang %}
```

Comment:
@3100 this is the messy part with the language URLs and why I ultimately went with `site.url` to solve it.

When viewing a translated lesson the `page.url` would be something akin to `/jp/lessons/basics/mix/` and relative links weren't doing the trick.  I also wanted to ensure the language links would go to the current lesson albeit translated:

1. If `page.lang` is set to something other than our default, remove the language code from the beginning of the `page.url`.  This leaves us with the section and lesson we're viewing. 

      > /jp/lessons/basics/mix -> /lessons/basics/mix

1. To create the links I concat `site.url` (our config value), `lang`, and the `url` from Step 1.